OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +136,38 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+			{
+				var guidedTargetPos = args.GuidedTarget.CenterPosition;
+				var targetDistance = new WDist((guidedTargetPos - args.Source).Length);
+
+				// Only continue tracking target if it's within weapon range +
+				// BeyondTargetRange to avoid edge case stuttering (start firing and immediately stop again).
+				if (targetDistance > args.Weapon.Range + info.BeyondTargetRange)
+				{
+					continueTracking = false;
+					isTailTravelling = true;
+					return;
+				}
+
+				target = guidedTargetPos;
+			}
+
+			towardsTargetFacing = (target - args.Source).Yaw.Facing;
+
+			// Update the target position with the range we shoot beyond the target by
+			// I.e. we can deliberately overshoot, so aim for that position
+			var dir = new WVec(0, -1024, 0).Rotate(WRot.FromFacing(towardsTargetFacing));
+			target += dir * info.BeyondTargetRange.Length / 1024;
+		}
+
 		public void Tick(World world)
 		{
+			if (info.TrackTarget && continueTracking)
```

Comment:
You can make this a bit cleaner by moving the `continueTracking` check inside `UpdateTarget` (it can tie in with the fix for the issue just above) and then rename the method to `TrackTarget`.  This then keeps all the logic related to target tracking in one place.